lib/xclarity_client/xclarity_base.rb

```diff
@@ -14,17 +14,18 @@ def initialize(conf, uri)
     end
 
     def connection_builder(conf, uri)
```

Comment:
I think here is necessary to check valid hostnames before mount the configurations.( e. g.: If the host is a empty string)